submissions/AriadnaKravchuk/opp-tiny-js-world/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/AriadnaKravchuk/a-tiny-JS-world
+   Web app: https://ariadnakravchuk.github.io/a-tiny-JS-world
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+
+const SPECIES = {
+   HUMAN: "human",
+   CAT: "cat",
+   DOG: "dog"
+}
+
+const GENDER = {
+   MALE: "male",
+   FEMALE: "female"
+}
+
+const SAYING = {
+   DOG: "woof-woof!",
+   CAT: "meow!"
+}
+
+class Inhabitant {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.friends = [];
+      this.attributes = [];
+   }
+
+   addFriends(...friend) {
+      for (let i = 0; i < friends.length; i++) {
+         if (!this.friends.includes(friends[i])) {
+            this.friends.push(friends[i]);
+         }
+      }
+   }
+
+   printAttributes() {
+      let friendsNames = [];
+
+      for (let i = 0; i < this.friends.length; i++) {
+         friendsNames.push(this.friends[i]["name"]);
+      }
```

Comment:
A classic use-case for `Array#map`. You need to recognize these as you will use Array methods heavily.

Also you can mix property reference notation `arrayOfObject[index].propName` is an equivalent to `arrayOfObject[index]["propName"]`. It is also better since IDE will be able to check if `.propName` exist, while property names as strings are not essentially testable.